packages/squiggle-lang/src/expression/compile.ts

```diff
@@ -126,6 +132,45 @@ function compileToContent(
         newContext,
       ];
     }
+    case "DecoratedStatement": {
+      // First, compile the inner statement.
+      // `newContext` includes the new binding.
```

Comment:
Quick note about this.

We transform "source code string" --(parse)--> "AST" --(compile)--> "Expression" --(reduce/evaluate)--> Result/Bindings.

For decorators, I could:
1. rewrite AST -> AST', where AST' won't have decorators, basically treating decorators as macros
2. or compile decorators AST to the expression that rewrites decorators as calls

I opted for (2), but (1) is also sometimes possible and might be useful in other situations. Sophisticated compilers often do an extra "optimization pass" that rewrites either AST or Expression (I'm not sure but I think it's mostly "Expressions", i.e. internal representation) for faster evaluation.

With decorators, because we have to distinguish between "decorator calls" from "normal calls", doing (1) would be more awkward.